embeddings/hyperparameter_search/configspace.py

```diff
@@ -0,0 +1,402 @@
+            return param.name, param.value
+        elif isinstance(param, ConstantParameter):
+            trial.set_user_attr(param.name, param.value)
+            return param.name, param.value
+        else:
+            raise ValueError(
+                f"Parameter type {type(param)} is not suported! "
+                f"Supported types are: {SearchableParameter.__name__} and {ConstantParameter.__name__}"
+            )
+
+    def _map_parameters(
+        self, trial: optuna.trial.Trial, parameters_names: List[str]
+    ) -> Dict[str, PrimitiveTypes]:
+        parameters: Dict[str, PrimitiveTypes] = {}
+        for param_name in parameters_names:
+            parameters.update([self._parse_parameter(trial=trial, param_name=param_name)])
+
+        return parameters
+
+    def _map_task_specific_parameters(
+        self, trial: optuna.trial.Trial
+    ) -> Tuple[Dict[str, PrimitiveTypes], Set[str]]:
+        return dict(), set()
+
+    @classmethod
+    def _get_fields(cls) -> Dict[str, Type[Parameter]]:
+        return {field_.name: field_.type for field_ in dataclasses.fields(cls)}
+
+    def sample_parameters(self, trial: optuna.trial.Trial) -> Dict[str, PrimitiveTypes]:
+        task_params, mapped_params_names = self._map_task_specific_parameters(trial)
+        params = self._map_parameters(
+            parameters_names=[
+                param_name
+                for param_name in self._get_fields().keys()
+                if param_name not in mapped_params_names
+            ],
+            trial=trial,
+        )
+        self._check_duplicated_parameters(params, task_params)
+        return {**params, **task_params}
+
+    @staticmethod
+    def _pop_parameters(
+        parameters: Dict[str, PrimitiveTypes], parameters_keys: Set[str]
+    ) -> Dict[str, PrimitiveTypes]:
+        return {k: parameters.pop(k) for k in parameters_keys if k in parameters}
+
+    @staticmethod
+    def _check_duplicated_parameters(*parameter_dicts: Any) -> Any:
+        assert not functools.reduce(set.intersection, (set(d.keys()) for d in parameter_dicts))
+
+    @staticmethod
+    def _check_unmapped_parameters(parameters: Dict[str, PrimitiveTypes]) -> None:
+        if len(parameters):
+            raise ValueError(
+                f"Some of the parameters are not mapped. Unmapped parameters: {parameters}"
+            )
+
+    @staticmethod
+    @abc.abstractmethod
+    def parse_parameters(parameters: Dict[str, PrimitiveTypes]) -> SampledParameters:
+        pass
+
+    @staticmethod
+    def _retrieve_embedding_type(embedding_name: str) -> str:
+        embedding = AutoFlairWordEmbedding.from_hub(repo_id=embedding_name)
+        if isinstance(embedding, FlairTransformerEmbedding):
+            embedding_type = "dynamic"
+        elif isinstance(embedding, StaticEmbedding):
+            embedding_type = "static"
+        else:
+            raise TypeError("Embedding type not recognized")
```

Comment:
likely should change to NameError or ValueError?
exception NameError: Raised when a local or global name is not found.
exception ValueError: Raised when an operation or function receives an argument that has the right type but an inappropriate value.

```suggestion
            raise NameError("Embedding type not recognized")
```

Reply:
I dont understand this remark.

I think that the TypeError is appropriate

 exception TypeError: Raised when an operation or function is applied to an object of inappropriate type. The associated value is a string giving details about the type mismatch.

Reply:
my way of thinking was that it can still be a valid embedding type but we do not handle that case hence -> nameerror but it is kind of ambiguous to me what error we should use here so maybe this is okay